api/envoy/service/tap/v2alpha/common.proto

```diff
@@ -48,23 +48,23 @@ message MatchPredicate {
     // The match configuration will always match.
     bool any_match = 4 [(validate.rules).bool.const = true];
 
-    // HTTP request match configuration.
-    HttpRequestMatch http_request_match = 5;
+    // HTTP request headers match configuration.
+    HttpHeadersMatch http_request_headers_match = 5;
```

Comment:
should we organize this a bit, just like how BufferedTrace organized? i.e.

```
message HttpMessageMatch {
  HttpHeadersMatch headers_match;
  HttpTrailersMatch trailers_match;
}
```
?

Reply:
This is how I had this originally, but IMO we don't want this, because it makes it much more confusing on how to apply and/or logic.